src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -470,14 +470,16 @@ impl EventProcessor {
 
             let source_window = xev.data.get_long(0) as xproto::Window;
 
-            // Equivalent to `(x << shift) | y`
-            // where `shift = mem::size_of::<c_short>() * 8`
+            // https://www.freedesktop.org/wiki/Specifications/XDND/#xdndposition
+            // "data.l[2] contains the coordinates of the mouse position relative to the root
+            // window."
+            // "data.l[2] = (x << 16) | y"
             // Note that coordinates are in "desktop space", not "window space"
             // (in X11 parlance, they're root window coordinates)
-            // let packed_coordinates = xev.data.get_long(2);
-            // let shift = mem::size_of::<libc::c_short>() * 8;
-            // let x = packed_coordinates >> shift;
-            // let y = packed_coordinates & !(x << shift);
+            let packed_coordinates = xev.data.get_long(2);
+            let x = (packed_coordinates >> 16) as i16;
+            let y = (packed_coordinates & 0xffff) as i16;
+            self.dnd.position = (x, y);
```

Comment:
Maybe store translated position directly? I don't think there's a reason to use _root_ coords?

Reply:
Changed it to store the translated position.